lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -31,7 +31,7 @@ sub _init {
     my ($self) = @_;
 
     return 0 unless my $job = $self->app->schema->resultset('Jobs')->find($self->param('testid'));
-    my $module = $job->modules->search({name => $self->param('moduleid')})->first;
+    my $module = $job->modules->find({name => $self->param('moduleid')});
```

Comment:
There's no unique constraint (only on a combination with other columns) so this should be limited to one row.

Reply:
What do you mean with "this should be limited to one row"? You want me to revert the change or did you misunderstand how "find" behaves when multiple matching rows are found? Because only the first row is returned though a warning is printed, see https://github.com/os-autoinst/openQA/pull/4202#discussion_r711798111 .

In this specific case, why are we ok to just pick the first "moduleid" match?

Reply:
Because in most tests the test module names are unique. However, if there's a test module with the same name in multiple categories this function wouldn't work very well. I have *not* designed this. To me it would make more sense if one would specify all the criteria to identify a test module here (category, name, script) or to actually have a unique constraint for only the name (or having to specify category/name and have a unique constraint only for both).

I would not clutter our logs with warnings about this. If you think this design flaw is worth improving, create a ticket. For the sake of this PR better just use query options like `{rows => 1, order_by => {-desc => 'id'}}` to ensure we get only one row.

Reply:
Should I use

```perl
search({name => $self->param('moduleid')}, {rows => 1, order_by => {-desc => 'id'}})->first;
```

https://metacpan.org/pod/DBIx::Class::ResultSet#find states "If no key is specified, the search is carried over all unique constraints which are fully defined by the available condition. If no such constraint is found, find currently defaults to a simple search->(\%column_values) […]. Note that this fallback behavior may be deprecated in further versions". So it would work and I consider it unlikely that DBIx will change that but we can be on the save side with search, right?